### PR TITLE
API changes for bulk retrieval of multiple states

### PR DESCRIPTION
**Pull request by dahlia**
This patch adjusts the out-most APIs so that application developers (i.e., @planetarium/9c-dev) can retrieve multiple account states through a single call.  Although the internals do not fully utilize the new interface, I will complete the internals in the next patch.

### CODE REVIEW

**Comment by dahlia** [comment]
@planetarium/libplanet Rebased.  Please take a look!

File: Libplanet/Blockchain/BlockChain.cs [hunk 1]
```diff
@@ -506,20 +540,25 @@ public Transaction<T> GetTransaction(TxId txId)
             FungibleAssetStateCompleter<T> stateCompleter = null
         )
         {
+            if (offset is null && Count < 1)
```
Comment (riemannulus) [inline Libplanet/Blockchain/BlockChain.cs]: Is `offset is null` condition for interface compability? IMO there is no reason to offset is nullable.

File: Libplanet/Store/Trie/MerkleTrie.cs [hunk 2]
```diff
@@ -108,14 +108,19 @@ public ITrie Set(in KeyBytes key, IValue value)
...
+        public IReadOnlyList<IValue?> Get(IReadOnlyList<KeyBytes> keys)
         {
-            return TryGet(
-                Root,
-                ImmutableArray<byte>.Empty,
-                ToKey(key),
-                out value);
+            var values = new IValue?[keys.Count];
+            int i = 0;
+            foreach (KeyBytes key in keys)
```
[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Store/Trie/MerkleTrie.cs:116]: Would it be okay if we assume every [`IReadOnlyList<T>`][1] implementation provides constant-time random access?

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.ireadonlylist-1?view=netstandard-2.0

[the remark this answers is not in the document]
Comment (riemannulus) [reply Libplanet/Store/Trie/MerkleTrie.cs:116]: [`IReadOnlyList<T>.Item[int32]`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.ireadonlylist-1.item?view=netstandard-2.0#system-collections-generic-ireadonlylist-1-item(system-int32)) is `IReadOnlyList<T>`'s interface. If `IReadOnlyList<T>`'s concrete class is vaild, we can use constant-time random access about `IReadOnlyList<T>`

[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Store/Trie/MerkleTrie.cs:116]: Fair enough.  [`LinkedList<T>`][1] apparently does not implement `IReadOnlyList<T>` at all.  If a type implements `IReadOnlyList<T>` without constant-time random access, that would be their fault.  I'm going to use `for` statement instead here.

[1]: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.linkedlist-1?view=netstandard-2.0

File: Libplanet/Blockchain/BlockChain.cs (hunk 1 above)
Comment (dahlia) [reply Libplanet/Blockchain/BlockChain.cs]: It was already fixed.  I guess you read the outdated revision.

File: Libplanet/Store/Trie/MerkleTrie.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dahlia) [reply Libplanet/Store/Trie/MerkleTrie.cs]: Amended!

**Comment by dahlia** [comment]
@riemannulus Applied your suggestions.  @libplanet Rebased.  Please take a look again!

**Review by riemannulus — APPROVED** [review]

**Comment by riemannulus** [comment]
👍

**Review by limebell — APPROVED** [review]

**Pull request merged**